src/mumble/Messages.cpp

```diff
@@ -1156,8 +1156,10 @@ void MainWindow::removeContextAction(const MumbleProto::ContextActionModify &msg
 ///
 /// @param msg The message object with the respective information
 void MainWindow::msgVersion(const MumbleProto::Version &msg) {
-	if (msg.has_version()) {
-		Global::get().sh->setProtocolVersion(msg.version());
+	if (msg.has_version_v2()) {
+		Global::get().sh->setProtocolVersion(msg.version_v2());
+	} else if (msg.has_version_v1()) {
+		Global::get().sh->setProtocolVersion(Version::fromLegacyVersion(msg.version_v1()));
```

Comment:
Reading the version from a `MumbleProto::Version` message might also be worth being wrapped in a function for the same reasons as writing the version...

Reply:
Can I add utility methods to the ``MumbleProto::Version`` object directly, or do I need to create a static or global function?

Reply:
As far as I can tell there is no (easy) way of adding additional methods to the proto objects themselves. So I guess a static function is our best option. I would probably embed these in the `MumbleProto` namespace such as this
```cpp
namespace MumbleProto {
    ::Version::mumble_raw_version_t getVersion(Version &versionMsg);
    void setVersion(Version &versionMsg, ::Version::mumble_raw_version_t);

    // Same for the SuggestConfig message
}
```

Reply:
In which file would I ideally put the implementation of these functions?

Reply:
Probably in a new file - `ProtoUtils` or something like that :eyes: